components/VoteReasons.tsx

```diff
@@ -39,6 +40,29 @@ export function VoteReasons({ vote }: VoteReasonProps) {
     return proposals.some(p => p.id === vote.proposal.id);
   }, [proposals, vote.proposal.id]);
 
+  const publicClient = usePublicClient();
+  const [nnsName, setNnsName] = useState<string | null>(null);
+
+  useEffect(() => {
+    const getNnsName = async () => {
+      try {
+        const calldata = '55ea6c47000000000000000000000000' + vote.voter.id.substring(2);
+        const res = await publicClient.call({
+          data: `0x${calldata}`,
+          to: '0x849f92178950f6254db5d16d1ba265e70521ac1b',
+        });
+        const offset = Number(BigInt(slice(res.data!, 0, 32)));
+        const length = Number(BigInt(slice(res.data!, offset, offset + 32)));
+        const name = fromHex(slice(res.data!, offset + 32, offset + 32 + length), 'string') || null;
+        // console.log(name);
+        setNnsName(name);
+      } catch (error) {
+        // console.log('Error getting NNS name:', error);
+      }
+    };
+    getNnsName();
+  }, []);
```

Comment:
missing reps here, vote.voter.id

Reply:
the deps is in the `calldata`

Reply:
will factor it into a method that takes `vote.voter.id` as an argument